src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
Wouldn't that then argue for going the other way, and just have both mongocxx::client and mongocxx::pool take only an options::ssl object, and kill of options::client entirely?